### PR TITLE
add with_about for CLI commands

### PR DESCRIPTION
**Pull request by Dominion5254**

### CODE REVIEW

File: core/startos/src/auth.rs [hunk 1]
```diff
@@ -91,29 +91,41 @@ pub fn auth<C: Context>() -> ParentHandler<C> {
                 .with_metadata("login", Value::Bool(true))
                 .no_cli(),
         )
-        .subcommand("login", from_fn_async(cli_login).no_display())
+        .subcommand(
+            "login",
+            from_fn_async(cli_login)
+                .no_display()
+                .with_about("Log in to StartOS server allowing use of tools such as side-loading services via CLI"),
```
Comment (dr-bonez) [inline core/startos/src/auth.rs:98]: This is maybe a bit too verbose - sideloading is just one of dozens of authenticated endpoints, and calling it out specifically here assumes too much about the intent of the user

File: core/startos/src/auth.rs [hunk 2]
```diff
@@ -290,13 +302,15 @@ pub fn session<C: Context>() -> ParentHandler<C> {
                 .with_custom_display_fn(|handle, result| {
                     Ok(display_sessions(handle.params, result))
                 })
-                .with_call_remote::<CliContext>(),
+                .with_call_remote::<CliContext>()
+                .with_about("Dispaly all server sessions"),
```
Comment (dr-bonez) [inline core/startos/src/auth.rs:306]: Dispaly! 😝

File: core/startos/src/lib.rs [hunk 3]
```diff
@@ -162,15 +172,26 @@ pub fn server<C: Context>() -> ParentHandler<C> {
...
                     Ok(system::display_time(handle.params, result))
                 })
-                .with_call_remote::<CliContext>(),
+                .with_call_remote::<CliContext>()
+                .with_about("Display current time and server uptime"),
+        )
+        .subcommand(
+            "experimental",
+            system::experimental::<C>()
+                .with_about("Experimental commands such as zram and governor"),
```
Comment (dr-bonez) [inline core/startos/src/lib.rs:181]: Maybe "Commands related to configuring experimental options such as zram and cpu governor"

File: core/startos/src/lib.rs [hunk 4]
```diff
@@ -222,19 +247,22 @@ pub fn server<C: Context>() -> ParentHandler<C> {
             .with_custom_display_fn(|_handle, result| {
                 Ok(firmware::display_firmware_update_result(result))
             })
-            .with_call_remote::<CliContext>(),
+            .with_call_remote::<CliContext>()
+            .with_about("Check for firmware updates and update if available"),
```
Comment (dr-bonez) [inline core/startos/src/lib.rs:251]: "Update the mainboards firmware to the latest firmware available in this version of StartOS if available. Note: This command does not reach out to the Internet"

File: core/startos/src/registry/db.rs [hunk 5]
```diff
@@ -18,14 +18,19 @@ use crate::util::serde::{apply_expr, HandlerExtSerde};
 
 pub fn db_api<C: Context>() -> ParentHandler<C> {
     ParentHandler::new()
-        .subcommand("dump", from_fn_async(cli_dump).with_display_serializable())
+        .subcommand(
+            "dump",
+            from_fn_async(cli_dump)
+                .with_display_serializable()
+                .with_about("Return db tables and records"),
```
Comment (Blu-J) [inline core/startos/src/registry/db.rs:25]: 👍 I'm glad that you where consistent with the previous db

**Review by Blu-J — APPROVED** [review]

File: core/startos/src/lib.rs (hunk 4 above)
Comment (Blu-J) [inline core/startos/src/lib.rs]: ❓ Didn't the original comment Aiden gave have a note, whey did we remove that. I thought it was useful
`Note: This command does not reach out to the Internet`

File: core/startos/src/auth.rs [hunk 6]
```diff
@@ -91,28 +91,40 @@ pub fn auth<C: Context>() -> ParentHandler<C> {
...
             from_fn_async(logout)
                 .with_metadata("get_session", Value::Bool(true))
                 .no_display()
+                .with_about("Log out of StartOS server")
                 .with_call_remote::<CliContext>(),
         )
-        .subcommand("session", session::<C>())
+        .subcommand(
+            "session",
+            session::<C>().with_about("List or Kill StartOS sessions"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/auth.rs]: Updated to lowercase 'kill'

File: core/startos/src/db/mod.rs [hunk 7]
```diff
@@ -31,16 +31,29 @@ lazy_static::lazy_static! {
 
 pub fn db<C: Context>() -> ParentHandler<C> {
     ParentHandler::new()
-        .subcommand("dump", from_fn_async(cli_dump).with_display_serializable())
+        .subcommand(
+            "dump",
+            from_fn_async(cli_dump)
+                .with_display_serializable()
+                .with_about("Return db tables and records"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/db/mod.rs]: Updated to "Filter/query db to display tables and records"

File: core/startos/src/db/mod.rs [hunk 8]
```diff
@@ -31,16 +31,29 @@ lazy_static::lazy_static! {
...
-        .subcommand("apply", from_fn_async(cli_apply).no_display())
+        .subcommand(
+            "put",
+            put::<C>().with_about("Command for adding UI record to db"),
+        )
+        .subcommand(
+            "apply",
+            from_fn_async(cli_apply)
+                .no_display()
+                .with_about("Update a db record"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/db/mod.rs:55]: The pattern I followed was to provide examples only in the case of additional subcommands exposed by a subcommand. I hadn't really considered providing specific examples for a more open ended subcommands like `apply` as providing a specific example here might assume too much of what the user could do with the `apply` command.

That said, I am open to providing examples in cases like this if we do want to go with that convention.

File: core/startos/src/lib.rs [hunk 9]
```diff
@@ -114,40 +114,90 @@ impl std::fmt::Display for ApiState {
...
+                "Commands related to Authentication such as logging in, resetting password, etc",
+            ),
+        )
+        .subcommand(
+            "db",
+            db::db::<C>().with_about("Commands to interact with the db i.e. dump, put, apply"),
+        )
+        .subcommand(
+            "ssh",
+            ssh::ssh::<C>().with_about("Add, delete, or list ssh keys"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/lib.rs]: Agreed on <short summary> followed by <example subcommands> as more helpful pattern.

Updated to "Commands for interacting with ssh keys i.e. add, delete, or list"

File: core/startos/src/lib.rs [hunk 10]
```diff
@@ -114,40 +114,90 @@ impl std::fmt::Display for ApiState {
...
+        )
+        .subcommand(
+            "wifi",
+            net::wifi::wifi::<C>().with_about("Commands related to wifi networks"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/lib.rs]: Agreed. Updated to "Commands related to wifi networks i.e. add, connect, delete"

File: core/startos/src/lib.rs [hunk 11]
```diff
@@ -329,74 +425,124 @@ pub fn package<C: Context>() -> ParentHandler<C> {
...
             from_fn(|_: DiagnosticContext| Ok::<_, Error>(ApiState::Error))
                 .with_metadata("authenticated", Value::Bool(false))
+                .with_about("Display the API that is currently serving")
                 .with_call_remote::<CliContext>(),
         )
-        .subcommand("diagnostic", diagnostic::diagnostic::<DiagnosticContext>())
+        .subcommand(
+            "diagnostic",
+            diagnostic::diagnostic::<DiagnosticContext>()
+                .with_about("Commands to display logs, restart the server, etc"),
```
[the remark this answers is not in the document]
Comment (Dominion5254) [reply core/startos/src/lib.rs]: Agreed. Updated to "Diagnostic commands i.e. logs, restart, rebuild"

File: core/startos/src/registry/db.rs (hunk 5 above)
Comment (Dominion5254) [reply core/startos/src/registry/db.rs]: Considering the update of the "dump" sub-command in core/startos/db/mod.rs I have updated this description to "Filter/query db to display tables and records" for consistency.

**Review by Blu-J — APPROVED** [review]

**Pull request merged**